openedx/core/lib/block_structure/transformer.py

```diff
@@ -125,4 +126,55 @@ def transform(self, usage_info, block_structure):
                 block structure, with already collected data for the
                 transformer, that is to be transformed in place.
         """
-        pass
+        raise NotImplementedError
+
+
+class OptimizedTransformer(BlockStructureTransformer):
```

Comment:
I'm confused by `use the most appropriate graph traversal algorithm for their needs.` Isn't the work we're doing on these combinable transformers forcing all of them to share the same graph traversal algorithm?

In any case, I'm on board with renaming this class - any great suggestions? `CombinableTransformer` seems a bit clunky, but it's the best I can come up with at the moment.